theano/gpuarray/corr_gemm.c

```diff
@@ -417,6 +417,11 @@ PyGpuArrayObject* corrMM(PyGpuArrayObject *const bottom,
                 "GpuCorrMM images and kernel must have the same stack size\n");
         return NULL;
     }
+    if ((nFilters %% numgroups) != 0) {
+        PyErr_SetString(PyExc_ValueError,
+                "GPUCorrMM the number of filters must be divisible by the number of groups\n");
+        return NULL;
+    }
```

Comment:
Yes, to explain in detail, input shape will be (batches, channels, ih,iw) and kernel shape will be (filters, channels/num_groups,kh,kw). So, in grouped convolution, each input shape will be (batches, channels/num_groups, ih, iw) and kernel shape will (_**filters/num_groups**_ , channels/num_groups, kh, kw). In other words, during the loop, the kernel is divided on nFilters dimension.